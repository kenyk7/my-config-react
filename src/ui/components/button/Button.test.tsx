import renderer from 'react-test-renderer';
import { Button } from '.';

it('Button basic primary renders correctly', () => {
  const component = renderer.create(<Button>Hello</Button>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('Button variant secondary renders correctly', () => {
  const component = renderer.create(<Button variant="secondary">Hello</Button>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('Button outlined renders correctly', () => {
  const component = renderer.create(<Button outlined>Hello</Button>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('Button block renders correctly', () => {
  const component = renderer.create(<Button block>Hello</Button>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('Button size renders correctly', () => {
  const component = renderer.create(<Button size="lg">Hello</Button>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('Button disabled renders correctly', () => {
  const component = renderer.create(<Button disabled>Hello</Button>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
