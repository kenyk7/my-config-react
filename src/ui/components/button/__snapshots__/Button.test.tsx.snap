// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button basic primary renders correctly 1`] = `
<button
  className="btn btn--primary"
  disabled={false}
  type="button"
>
  Hello
</button>
`;

exports[`Button block renders correctly 1`] = `
<button
  className="btn btn--primary btn--block"
  disabled={false}
  type="button"
>
  Hello
</button>
`;

exports[`Button disabled renders correctly 1`] = `
<button
  className="btn btn--primary btn--disabled"
  disabled={true}
  type="button"
>
  Hello
</button>
`;

exports[`Button outlined renders correctly 1`] = `
<button
  className="btn btn--primary btn--outlined"
  disabled={false}
  type="button"
>
  Hello
</button>
`;

exports[`Button size renders correctly 1`] = `
<button
  className="btn btn--lg btn--primary"
  disabled={false}
  type="button"
>
  Hello
</button>
`;

exports[`Button variant secondary renders correctly 1`] = `
<button
  className="btn btn--secondary"
  disabled={false}
  type="button"
>
  Hello
</button>
`;
